Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt

```diff
@@ -0,0 +1,103 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules.keysubmission
+
+import de.rki.coronawarnapp.datadonation.analytics.common.toMetadataRiskLevel
+import de.rki.coronawarnapp.datadonation.analytics.storage.AnalyticsSettings
+import de.rki.coronawarnapp.risk.RiskLevelSettings
+import de.rki.coronawarnapp.risk.storage.RiskLevelStorage
+import de.rki.coronawarnapp.risk.tryLatestResultsWithDefaults
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData
+import de.rki.coronawarnapp.util.TimeStamper
+import kotlinx.coroutines.flow.first
+import org.joda.time.Duration
+import javax.inject.Inject
+
+class AnalyticsKeySubmissionCollector @Inject constructor(
+    private val timeStamper: TimeStamper,
+    private val analyticsSettings: AnalyticsSettings,
+    private val analyticsKeySubmissionStorage: AnalyticsKeySubmissionStorage,
+    private val riskLevelStorage: RiskLevelStorage,
+    private val riskLevelSettings: RiskLevelSettings
+) {
+
+    fun reset() {
+        analyticsKeySubmissionStorage.clear()
+    }
+
+    fun reportPositiveTestResultReceived() {
+        if (isEnabled) analyticsKeySubmissionStorage.testResultReceivedAt.update { timeStamper.nowUTC.millis }
```

Comment:
Not a blocker, but personally, I would prefer a simple check using the same pattern everywhere, no double negations, or different checks of `isEnabled` vs `isDisabled`, i.e. early aborts with the same pattern.

Everywhere just `if(!isEnabled) return`

Reply:
I prefer if (disabled). Negation is so hard to read.